platform/feature-api/api/src/pod-api/api.ts

```diff
@@ -106,21 +105,50 @@ export interface Entry {
     path: string;
 }
 
+export interface EncodingOptions {
+    encoding: BufferEncoding;
+}
+
+export interface Stats {
+    id: string;
+    size: number;
+    time: string;
+    name: string;
+    directory: boolean;
+}
+
 /**
  * `PolyOut` specifies the interaction of the Feature with the environment. It is concerned with file system operations
  * and HTTP requests.
  *
  * Both of these aspects are separated out into their own modules:
  * - [[FS]] for Node.js-style file-system access
  */
-export interface PolyOut extends Omit<FS, "readdir"> {
+export interface PolyOut {
+    readFile(path: string, options: EncodingOptions): Promise<string>;
+    readFile(path: string): Promise<Uint8Array>;
+    writeFile(
+        path: string,
+        content: string,
+        options: EncodingOptions
+    ): Promise<void>;
```

Comment:
@fhd Is the `options` argument to `readFile` and  `writeFile` implemented or planned to be implemented anywhere, or is that as well just an artifact due to the inspiration from the Node.js FS API? In that case I'm happy to remove the `options` argument as well from the interface.

Reply:
As far as I can tell, just an artifact. In fact, there was no plans to implement `writefile`, in general.

Reply:
Here we go: #1137